app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  before_action :ensure_enabled
+
+  def index
+    scope = current_project.present? ? stage.deploy_groups : DeployGroup.all
+    render json: paginate(scope.sort_by(&:natural_order))
```

Comment:
wat? paginate breaks them into clumps of 1000.

if there is less than 1000, then yes, paginate is pointless, but >1000 and it doesn't show them all.